pkg/dochandler/didvalidator/validator.go

```diff
@@ -87,7 +87,7 @@ func (v *Validator) IsValidOriginalDocument(payload []byte) error {
 	// Sidetree rule: add service validation
 
 	// generic did document validation - must have context
-	if didDoc.Context() != didContext {
+	if didDoc.Context() != nil && didDoc.Context()[0] != didContext {
```

Comment:
suggest calling didDoc.Context() once and assigning to variable first. Check that the variable's length != 0 (rather than nil check).